File: pset6/pset6/cash/cash.py
from cs50 import get_float

while True:
    change = get_float("Change owed: ")

    if change > 0:
        break

#print(f"{change}") # gets inputed change
convertedChange = round(change * 100); #needed to avoid floating point imprecision which causes errors in calculation

'''
converte-se o troco para evitar erros de calculo
e depois a esse troco ve-se quantas vezes se pode
retirar uma moeda , neste caso, de 0.25.
o numero de x q se puder retirar cada moeda é = ao
numero de moedas necessario p cobrir esse valor
'''
coinsCounter = 0

while convertedChange >= 25:
   convertedChange = convertedChange - 25
   coinsCounter = coinsCounter + 1
   #print(coinsCounter) # 40 moedas
#print(coinsCounter)


while convertedChange <= 25 and convertedChange >= 10:
   convertedChange = convertedChange - 10
   coinsCounter = coinsCounter + 1
#print(coinsCounter)

while convertedChange <= 10 and convertedChange >= 5:
   convertedChange = convertedChange - 5
   coinsCounter = coinsCounter + 1

while convertedChange <= 5 and convertedChange >= 1:
   convertedChange = convertedChange - 1
   coinsCounter = coinsCounter + 1
print(coinsCounter)

File: pset1/pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main()
{
    float change;
    int convertedValue; //needed to avoid floating imprecision which causes errors in calculation
    int qtyOfCoins = 0;

    do
    {
        change = get_float("How much change is owed? \n");
        convertedValue = round(change * 100); //needed to avoid floating point imprecision which causes errors in calculation
    }

    while (change < 0);

    //printf("%.2f", change);
    //printf("\n");
    while (convertedValue >= 25)
    {
        convertedValue = convertedValue - 25;
        qtyOfCoins++;

    }
    while (convertedValue < 25 && convertedValue >= 10)
    {
        convertedValue = convertedValue - 10;
        qtyOfCoins++;
    }
    while (convertedValue < 10 && convertedValue >= 05)
    {
        convertedValue = convertedValue - 5;
        qtyOfCoins++;
    }

    while (convertedValue < 5 && convertedValue > 0)
    {
        convertedValue = convertedValue - 1;
        qtyOfCoins++;
    }
    printf("%i", qtyOfCoins);
    printf("\n");
}